### DISCUSSION TITLE
Permission errors when using make in docker build file

### DISCUSSION THREAD

**Discussion by cleethai** [edenhill/librdkafka#3426] [Q&A]
I am very new to docker and librdkafka, but I have inherited a setup from a year ago.  I am working on an Ubuntu 18.04 system with Docker Version: 19.03.14.

I have a very simple docker build file:
```
FROM alpine:edge

RUN apk update && \
    apk add --no-cache \
    git \
    gcc \
    g++ \
    make \
    bash \
    go \
    pkgconfig \
    openssh-client 

RUN git clone --depth=1 --single-branch --branch=master https://github.com/edenhill/librdkafka.git && \
    cd librdkafka && \
    ./configure --prefix /usr && \
    make -j4 && \
    make install
```
which is called from a build script ran with sudo permissions:
```
#!/bin/bash
set -e
set -o pipefail

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"

function checkDocker() {
  docker info >/dev/null 2>&1 || {
    echo -e "\nseems like you dont have enough permission :(";
    echo -e "try sudo perhaps?\n";
    exit 2
  }
}

checkDocker

docker build -t image-cap19-builder $DIR/
```

When I run the build script, I get the following error when the docker build hits the make: 
```
make: uname: Operation not permitted
make: /bin/sh: Operation not permitted
make: *** [mklove/Makefile.base:91: mklove-check] Error 127
```

If I run all the commands outside the docker build, inserting sudo in front of the makes, it works, but installing and inserting sudo in front in the docker build file doesn't work.

Anyone have any ideas? 

Thanks.

**Comment by DuCNiKo** [comment]
Hi, I've got exactly the same issue :) 

It's because of [that](https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.14.0#faccessat2)... Honestly, I'm a .Net developper and not an expert on Linux GNU & other things like that... But from what I've understood : 

The change between Alpine 3.13 & 3.14 about the fsaccessat2 is totally transparent IF : 
* you use a recent version of docker (not my case, I'm struck in 19.0.something)
* you have installer a runc package

Or you can simply (and for now) use an alpine 3.13 instead of alpine:edge... It worked for me... 
I know it's not a long term solution but at least for now it works... 

Regards,
